### PR TITLE
49 geoquery microarray studies

### PR DESCRIPTION
**Pull request by alexgrent**
Fetching data from GEO, via Python and R within the DatasetFetchers

### CODE REVIEW

**Review by jgriss — CHANGES_REQUESTED** [review]
Hi @alexgrent !

This looks great! Only minor points we could discuss - but as you saw I'm not 100% sure myself about them.

Let me know what you think, then I'll merge everything and run the integration tests.

File: reactome_analysis_api/reactome_analysis_api/controllers/datasets_controller.py [hunk 1]
```diff
@@ -83,7 +83,16 @@ def get_data_sources():  # noqa: E501
                            parameters=[
                                ExternalDatasourceParameters(name="dataset_id", display_name="Dataset Id",
                                                             type="string", description="Identifier of the dataset",
-                                                            required=True)])
+                                                            required=True)]),
+        ExternalDatasource(id="geo", name="GEO expression",
```
[the remark this answers is not in the document]
Comment (alexgrent) [reply reactome_analysis_api/reactome_analysis_api/controllers/datasets_controller.py:87]: changed the id to  geo_mircroarray legacy sounds a little bit prejudice

**Comment by alexgrent** [comment]
Hi changed according to review. 
I think you are right we can differentiate a little bit more to GREIN, 
changed the id and the description to be more specific to microarray data

**Comment by alexgrent** [comment]
squash commits before merging

**Review by jgriss — APPROVED** [review]
@alexgrent Thanks a lot! Looks great!

**Pull request merged**